src/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1486,5 +1499,112 @@ private bool TryGetHostUri(string hostName, out Uri hostUri)
+                handler.Credentials = parameters.Credentials;
+                handler.AllowAutoRedirect = parameters.AllowAutoRedirect;
+                handler.MaxAutomaticRedirections = parameters.MaximumAutomaticRedirections;
+                handler.MaxResponseHeadersLength = parameters.MaximumResponseHeadersLength;
+                handler.PreAuthenticate = parameters.PreAuthenticate;
+                client.Timeout = parameters.Timeout;
+
+                if (parameters.CookieContainer != null)
+                {
+                    handler.CookieContainer = parameters.CookieContainer;
+                    Debug.Assert(handler.UseCookies); // Default of handler.UseCookies is true.
+                }
+                else
+                {
+                    handler.UseCookies = false;
+                }
+
+                Debug.Assert(handler.UseProxy); // Default of handler.UseProxy is true.
+                Debug.Assert(handler.Proxy == null); // Default of handler.Proxy is null.
+
+                // HttpClientHandler default is to use a proxy which is the system proxy.
+                // This is indicated by the properties 'UseProxy == true' and 'Proxy == null'.
+                //
+                // However, HttpWebRequest doesn't have a separate 'UseProxy' property. Instead,
+                // the default of the 'Proxy' property is a non-null IWebProxy object which is the
+                // system default proxy object. If the 'Proxy' property were actually null, then
+                // that means don't use any proxy.
+                //
+                // So, we need to map the desired HttpWebRequest proxy settings to equivalent
+                // HttpClientHandler settings.
+                if (parameters.Proxy == null)
+                {
+                    handler.UseProxy = false;
+                }
+                else if (!object.ReferenceEquals(parameters.Proxy, WebRequest.GetSystemWebProxy()))
+                {
+                    handler.Proxy = parameters.Proxy;
+                }
+                else
+                {
+                    // Since this HttpWebRequest is using the default system proxy, we need to
+                    // pass any proxy credentials that the developer might have set via the
+                    // WebRequest.DefaultWebProxy.Credentials property.
+                    handler.DefaultProxyCredentials = parameters.Proxy.Credentials;
+                }
+
+                if (parameters.ClientCertificates != null)
+                {
+                    handler.ClientCertificates.AddRange(parameters.ClientCertificates);
+                }
+
+                // Set relevant properties from ServicePointManager
+                handler.SslProtocols = (SslProtocols)ServicePointManager.SecurityProtocol;
+                handler.CheckCertificateRevocationList = ServicePointManager.CheckCertificateRevocationList;
```

Comment:
Shouldn't both of these be using the values from parameters, not from ServicePointManager?

Reply:
It was a mistake. Fixed.